### PR TITLE
DOM update 2024-09-25

### LINKED ISSUE #59704: `lib.d.ts` Updates for TypeScript 5.7
Including

* https://github.com/microsoft/TSJS-lib-generator/
* Outstanding `lib.d.ts` PRs to TS

### PR DESCRIPTION
**Pull request by sandersn**
This update is almost all additions; the deprecated MutationEvent is removed, and some events have more specific types. I think this is a low-impact update.

- new interface AddressErrors, PayerErrors, PaymentOptions, PaymentShippingOption, PaymentAddress, plus new properties in existing related types
- new interfaces AudioData*, AudioDecoder*, AudioEncoder*, EncodedAudioChunk*, VideoEncoder*
- various new properties in VideoConfiguration, VideoEncoder*
- new interfaces AuthenticationExtensionPRFInputs/Outputs/Values, plus associated `prf` properties in AuthenticationExtensionsClientInputs/Outputs
- new interface CarePositionFromPointOptions
- delete interface IntersectionObserverEntryInit
- various Media interfaces add `backgroundBlur` properties
- PointerEventInit adds `altitudeAngle` and `azimuthAngle`
- new interface CaretPosition
- new interfaces PublicKeyCredential*, plus associated new methods
- ARIAMixin adds properties ariaColIndexText, ariaRowIndexText
- AnimationEventMap's remove event now has type AnimationPlaybackEvent, not just Event
- AudioWorkletNodeEventMap's error event now has type ErrorEvent, not just Event
- HTMLVideoElementEventMap's events now have specific type PictureInPictureEvent
- MIDIAccessEventMap's event now have specific type MIDIConnectionEvent
- MediaRecorderEventMap's error event now has type ErrorEvent
- RTCDataChannelEventMap's error event now has type RTCErrorEvent
- De/CompressionStream is no longer an empty subtype of GenericTransformStream; adds ReadableStream and WriteableStream properties
- Document adds properties `fragmentDirective`, `caretPositionFromPoint`, `currentCSSZoom`
- deprecated interface MutationEvent is removed; Document.createEvent removes MutationEvent/s overloads.
- FontFace's constructor takes `source: BufferSource` instead of `source: BinaryData`
- FontFaceSet's contructor no longer has `initialFaces: FontFace[]` parameter.
- HTMLDetailsElement adds a property `writingSuggestions` 🤖 
- IntersectionObserverEntry's constructor no longer has `init` parameter.
- new interface ViewTransitionUpdateCallback

Fixes #59704

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot test it

**Comment by sandersn** [comment]
DT errors:

Polyfills that need to be updated:
- webrtc
- dom-webcodecs
- dom-mediacapture-record

A user of FontFaceSet that needs to update
- css-font-loading-module

A user of MutationEvent that needs a shim now, I guess?
- rx-dom

Node has a broken ExpectType: `ReadableStream<Uint8Array<ArrayBufferLike>>` 
- node

File: src/lib/dom.generated.d.ts [hunk 1]
```diff
@@ -450,6 +540,18 @@ interface ElementDefinitionOptions {
     extends?: string;
 }
 
+interface EncodedAudioChunkInit {
+    data: AllowSharedBufferSource;
+    duration?: number;
```
Comment (jakebailey) [inline src/lib/dom.generated.d.ts:545]: Maybe this is generically wrong with the DOM types, but this is seemingly an options bag given to a constructor. Why don't all of these have `| undefined`?

**Comment by jakebailey** [comment]
Hm, 26 new errors in the angular benchmark. Maybe that will show in the top tests, but it sounds like I should add that as a user test.

**Comment by jakebailey** [comment]
Retesting user tests now that I added angular there explicitly:

@typescript-bot user test this

File: src/lib/dom.generated.d.ts (hunk 1 above)
Comment (sandersn) [reply src/lib/dom.generated.d.ts:545]: I'd have to go read the code but I'm pretty sure it's an oversight that the generator never got updated for the missing/undefined distinction.

**Comment by sandersn** [comment]
I'll send PRs for the DT failures later today. 

The angular result for the most recent user test run says 36 errors before, 36 after.

**Review by jakebailey — APPROVED** [review]

**Pull request merged**